tests/test_security.py

```diff
@@ -0,0 +1,72 @@
+import pytest
+from fastapi import HTTPException
+
+from app.api.security import verify_token
+
+
+def test_missing_client_id_raises_error_when_auth_enabled(
+    monkeypatch, test_app
+):
+    """Test that a missing client ID raises an error on startup when authentication is enabled."""
+    # We're using what should be default values of CLIENT_ID and AUTH_ENABLED here
+    # (if the corresponding environment variables are unset),
+    # but we set the values explicitly here for clarity
+    monkeypatch.setattr("app.api.security.CLIENT_ID", None)
+    monkeypatch.setattr("app.api.security.AUTH_ENABLED", True)
+
+    with pytest.raises(ValueError) as exc_info:
+        with test_app:
+            pass
+
+    assert "NB_QUERY_CLIENT_ID is not set" in str(exc_info.value)
+
+
+# Ignore startup warning that is unrelated to the current test
+@pytest.mark.filterwarnings(
+    "ignore:No local Neurobagel nodes defined or found"
+)
+def test_missing_client_id_ignored_when_auth_disabled(monkeypatch, test_app):
+    """Test that a missing client ID does not raise an error when authentication is disabled."""
+    monkeypatch.setattr("app.api.security.CLIENT_ID", None)
+    monkeypatch.setattr("app.api.security.AUTH_ENABLED", False)
+
+    with test_app:
+        pass
+
+
+@pytest.mark.parametrize(
+    "invalid_token",
+    ["Bearer faketoken", "Bearer", "faketoken", "fakescheme faketoken"],
```

Comment:
where do these examples come from? They seem confusingly specific, but then I am not sure if the only reason they fail is because they are just random strings.

Reply:
e.g. what is the significance of the space in the first and last example?

Reply:
> the only reason they fail is because they are just random strings.

this is the main reason they fail. My goal here was just to give different examples of combinations of clearly invalid string tokens (`faketoken`) with our 'expected' token scheme `Bearer`.

In theory, the 'correct' authorization header that should lead to successful token validation in the f-API is
`Authorization: Bearer <validtokenfromGoogleauth>`

The `verify_token` function examines the part after `:`, and extracts `<validtokenfromGoogleauth>`. So the second and third examples should fail because they are missing either the scheme or the token value. 

The first example should fail because the `<validtokenfromGoogleauth>` part is invalid / just a random string, and the last example should technically fail because 1) the token scheme is not `Bearer`, 2) the `<validtokenfromGoogleauth>` part is invalid. However, `verify_token` currently doesn't check that the scheme is `Bearer` (hence the `TODO` there), so the last example should only be failing because of reason 2) atm.